File: leetcode/removeDuplicates.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = 1
        n = len(nums)
        if n == 0:
            return n
        for i in range(n-1):
            if nums[i] != nums[i+1]:
                nums[l] = nums[i+1]
                l += 1
        return l

File: leetcode/removeDuplicates.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.empty())
            return 0;
        int n = nums.size();
        int l = 1;
        int i = 0;
        for(i = 0; i < n - 1; ++i) {
            if(nums[i] != nums[i + 1]) {
                nums[l] = nums[i + 1];
                l++;
            }
        }
        return l;
    }
};